src/url.ts

```diff
@@ -118,24 +118,34 @@ export class Url {
     return {...(this.parts.params || {})};
   }
 
-  get nonNullParams(): Record<string, string> {
-    return Object.keys(this.parts.params || {}).reduce<Record<string, string>>((memo, key) => {
-      if (this.parts.params![key] != null) {
-        memo[key] = this.parts.params![key]!.toString();
-      }
+  get nonNullParams(): Record<string, string[]> {
+    const params = this.parts.params || {};
+
+    return Object.keys(params).reduce<Record<string, string[]>>(
+      (memo, key) => {
+        const value = params[key];
+
+        if (value != null) {
+          memo[key] ? memo[key].push(value.toString()) : (memo[key] = [value.toString()]);
+        }
 
-      return memo;
-    }, {});
+        return memo;
+      },
+      {} as Record<string, string[]>
+    );
   }
 
   get paramsAsString(): string {
     return Object.keys(this.params || {})
       .sort((a, b) => (a < b ? -1 : a === b ? 0 : 1))
       .reduce((memo: string[], key: string) => {
-        if (this.params[key] != null) {
-          memo.push(`${key}=${encodeURIComponent(this.params[key]!.toString())}`);
+        const value = this.params[key];
+
+        if (value != null) {
+          const values = value instanceof Array ? value : [value];
+          values.forEach(valueItem => memo.push(`${key}=${encodeURIComponent(valueItem.toString())}`));
         } else {
-          memo.push(`${key}`);
+          memo.push(key);
```

Comment:
I guess same here, `key` should be encoded, I guess we need to do it only in one line